include/GIL/Module.hpp

```diff
@@ -0,0 +1,56 @@
+#ifndef GLU_GIL_MODULE_HPP_
+#define GLU_GIL_MODULE_HPP_
+
+#include <list>
+#include <string>
+#include <memory>
+
+
+namespace glu::gil {
+
+class Function;
+
+class Type;
+
+/// @brief This class represents modules in our gil intermediate representation
+/// Modules are used to automaticaly import all delcarations in a file,
+/// see https://glu-lang.org/modules/
+class Module {
+public:
+    /// @brief Constructor for a Module instance that initialise its name
+    /// @param moduleName A string representing the modules name
+    Module(std::string moduleName);
+
+    /// @brief Adds a Function to the Functions dedicated chained list
+    /// @param function A pointer to the function that will be added
+    void addFunction(Function *function);
```

Comment:
I think the module should own the functions. Should maybe have unique pointers, OR store the instance directly and have a createFunction instead.

Reply:
I did it with unique_pointers first but It's not possible with forward declaration, so I guess owning the function is the best way to go.

Reply:
we will be able to #include <Function> soon.